### PR TITLE
simple user CRUD

### PR DESCRIPTION
**Pull request by EKo3t**

### CODE REVIEW

File: src/main/java/shadow/dictionary/api/model/DeleteUserRequest.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+
+@Getter
+public class DeleteUserRequest {
+
+    private final String login;
+
+    @JsonCreator
+    public DeleteUserRequest(@JsonProperty("login") String login) {
```
Comment (mgelios) [inline src/main/java/shadow/dictionary/api/model/DeleteUserRequest.java:13]: I would rather change "login" to "username" due to ambiguity of the "login" term (like username and log-in action)

File: src/main/java/shadow/dictionary/model/User.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+@NoArgsConstructor
+@Getter
+@Setter
+public class User {
+
+    @Id
+    private String id;
+    @Indexed(unique = true)
+    private String login;
```
Comment (mgelios) [inline src/main/java/shadow/dictionary/model/User.java:19]: same suggestion - login to username/nickname/name renaming

File: src/main/java/shadow/dictionary/service/UserCrud.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+import shadow.dictionary.api.model.NewUserRequest;
+import shadow.dictionary.api.model.UpdateUserRequest;
+import shadow.dictionary.dao.UserRepository;
+import shadow.dictionary.exception.UserNotFoundException;
+import shadow.dictionary.model.User;
+
+@Service
+public class UserCrud {
```
Comment (mgelios) [inline src/main/java/shadow/dictionary/service/UserCrud.java:13]: Maybe UserService for more common naming?

**Review by mgelios — CHANGES_REQUESTED** [review]
Suggested small renaming of the classes/fields. Take a look, please :)

File: src/main/java/shadow/dictionary/api/model/DeleteUserRequest.java (hunk 1 above)
Comment (EKo3t) [reply src/main/java/shadow/dictionary/api/model/DeleteUserRequest.java]: done

File: src/main/java/shadow/dictionary/model/User.java (hunk 2 above)
Comment (EKo3t) [reply src/main/java/shadow/dictionary/model/User.java]: done

File: src/main/java/shadow/dictionary/service/UserCrud.java (hunk 3 above)
Comment (EKo3t) [reply src/main/java/shadow/dictionary/service/UserCrud.java]: done

**Review by mgelios — APPROVED** [review]

**Pull request merged**